apps/src/applab/applab.js

```diff
@@ -280,10 +281,26 @@ function outputApplabConsole(output) {
 
 var apiWarn = outputApplabConsole;
 
-function apiValidateType(opts, funcName, varName, varValue, expectedType) {
+var OPTIONAL = true;
+
+function apiValidateType(opts, funcName, varName, varValue, expectedType, opt) {
```

Comment:
If you wanted to reduce redundancy, at the expense of some flexibility, you could omit the varValue param and set it as "var varValue = opts[varName]"